lib/board.rb

```diff
@@ -0,0 +1,85 @@
+require 'pry'
+class Board
+  attr_accessor :cells, :free_corner
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+
+  # reader method for board cells
+  def cells
+    @cells
+  end
+
+  # resets the board cell for a new game?
+  def reset!
+    cells.clear
+    self.cells = Array.new(9, " ")
+  end
+
+  # method to display the game board. not the same as the cell array.
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  # convert input to idx
+  def idx(input)
+    @idx = input.to_i - 1
+  end
+
+  # position method already has #idx built in so dont need to call #idx again.
+  def position(input)
+    cells[self.idx(input)]
+  end
+
+
+  def full?
+    cells.all?{|cell| cell != " "}
+  end
+
+  def turn_count
+    cells.select{|cell| cell != " "}.count
+  end
+
+  def taken?(input)
+    position(input) == " " ? false : true
+  end
+
+  def valid_move?(input)
+    if (0..8).include?(self.idx(input)) && self.taken?(input) == false
```

Comment:
You should be able to use the implicit return values rather than check for equality. 

`input.to_i.between?(1,9) && !taken(input)`